snowsql/resource_exec.go

```diff
@@ -136,7 +169,18 @@ func resourceExecRead(ctx context.Context, d *schema.ResourceData, m interface{}
 func resourceExecUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	// https://github.com/aidanmelen/terraform-provider-snowsql/issues/11
+	// TODO read before update
+
+	if d.HasChange("update.0.statements") {
```

Comment:
Do you have any opinions about this change?

Reply:
I'm fine with that! Much easier to read.

My reasoning was that update wouldn't even be called if it didn't change, but update what also be called if the delete lifecycle changed - and then, it's unnecessary to re-run the update script. So even cleaner as well.